synapse/lib/nexus.py

```diff
@@ -92,30 +94,83 @@ async def __anit__(self, dirn: str, donexslog: bool = True, map_async=False):  #
         self._futures: Dict[str, asyncio.Future] = {}
 
         path = s_common.genpath(self.dirn, 'slabs', 'nexus.lmdb')
+        fresh = not os.path.exists(path)
+
+        logpath = s_common.genpath(self.dirn, 'slabs', 'nexuslog')
 
         self.map_async = map_async
         self.nexsslab = await s_lmdbslab.Slab.anit(path, map_async=map_async)
-
-        self.nexslog = self.nexsslab.getSeqn('nexuslog')
         self.nexshot = await self.nexsslab.getHotCount('nexs:indx')
 
+        if fresh:
+            self.nexshot.set('version', 2)
+
+        vers = self.nexshot.get('version')
+
+        if vers <= 1:
+            await self._migrateV1toV2(path, logpath)
+        elif vers != 2:
```

Comment:
This may suffer from the same re-upgrade logic and never be hit?

Reply:
Same answer: protection from versions in the future.